### ISSUE TITLE
[webcontent] Miserable performance on mobile.twitter.com

### ISSUE THREAD

**Issue by j1elo** [mozilla-mobile/fenix#18944]
This is a repost of this Bugzilla issue:

**Miserable performance on mobile.twitter.com** (https://bugzilla.mozilla.org/show_bug.cgi?id=1447419)

That report didn't get any meaningful discussion before being automatically closed by a bot (arguing that "*The development of the new versions use GitHub for issue tracking. If the bug report still reproduces [...] an issue can be reported at the Fenix GitHub project*").

There are however similar bugs such as https://github.com/mozilla-mobile/fenix/issues/17229 which are being moved from here back to Bugzilla, so it's pretty confusing to follow up on *where* these bugs should be actually handled.

Other reports that might be related to this one are:
* Weird loading issues on Twitter (https://github.com/mozilla-mobile/fenix/issues/17144)
* Twitter behaving weirdly on firefox (https://github.com/mozilla-mobile/fenix/issues/17229)
* Scrolling is bad (https://github.com/mozilla-mobile/fenix/issues/18640)

## Site with issue and any steps to reproduce

I am reproducing the original report, because it describes the issue perfectly (including a screen recording), which is still happening as of today after testing with a clean installation of [Firefox Nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix):

> mobile.twitter.com's performance is absolutely miserable in Android Firefox. Running Beta 60.0 from the Play Store on a Galaxy S8 Edge, connected to an enterprise-grade wifi AP on gigabit internet.
> 
> Initial page loads are pretty fast, and scrolling doesn't usually hitch up, but after the tab has been open for around a minute or more it typically takes 500ms+ to respond to any taps. Text input is even slower, I can easily type a word or two on the OS keyboard before any of it shows up in their text fields. The page layout also jitters around a lot on a very long delay - I'll scroll down, tap a retweet button, and watch for 2-3 seconds as the page layout jitters around and then it registers my tap on an entirely separate part of the page because the layout changed. I also frequently see blank areas when scrolling, and image embeds in tweets can sometimes take multiple seconds to appear once they scroll into view (in the interim they're solid colored boxes.)
> 
> I wouldn't be surprised if 100% of this is Twitter bugs but it's pretty nasty. In particular it seems awful that taps can go to a different element if the page layout changes in the 2000ms between my tap and the actual processing of the event - the compositor knows where the layers and/or elements were when I tapped, right?
> 
> I tried checking about:performance for info but it never finishes loading on Android.
> 
> Here's a screen recording: https://www.youtube.com/watch?v=JIVUKpmI9DI
> 
> When I tapped the like button a couple times each one took upwards of 1000ms to register. You can see slow loading of images and blank gaps as well. This is a fresh load of a mobile twitter tab.

### Expected behavior

Smooth(er) movements, and a minimum of reliability in the processing of which elements are touched by the user, and which ones end up receiving the corresponding events.

### Actual behavior

Lots of small unexpected issues, leading to a broken user experience: glitches on scrolling, exceedingly large loading times (regardless of quality of network connection), unreliable touch events.

In summary, if I had to make a screen recording showing the issues with the latest Firefox Nightly version as of today, it would look pretty much the same as the one that was provided in the original report.

### Does toggling Tracking Protection fix the issue? (Press the shield icon in the toolbar while on the site to see toggle)

No.

### Can you reproduce in Chrome (or other non-Mozilla browser)?

No. While Twitter's UI responsiveness is far from ideal regardless of web browser, Chrome shows minimal issues with loading images, tweet contents, and overall interface elements.

### Device information

* Android device: Xiaomi Redmi Note 4X, [PixelExperience](https://download.pixelexperience.org/about) with Android 9.
* Fenix version: `version Nightly 210411 17:01`

**Comment by jonalmeida** [comment]
The bugzilla issue you linked was using Fennec, an old version of Firefox for Android that is no longer supported.

There is already a new issue opened in bugzilla that was moved from another github issue, so this one here can be closed as a duplicate of it: https://bugzilla.mozilla.org/show_bug.cgi?id=1685629

**Comment by Mugurell** [comment]
@j1elo 
> why then the announcement about moving everything to GitHub in the first place?

The development of the application is now managed on Github while previously it was done in Bugzilla.
But Gecko and GeckoView development is still managed on Bugzilla.

So issues with the Android application are tracked on Github (how the app as a GeckoView wrapper looks and behaves) but issues related to webpages and website interactions are tracked on Bugzilla as they are browsing issues and not app issues and that is where Gecko development is tracked.

**Comment by j1elo** [comment]
Thanks for the clarification, from a developer point of view this distinction makes sense. It makes me think though, how many good reports related to Gecko and GeckoView, thus unrelated to the specific issues of the old application, might have been lost (automatically closed) with the migration to GitHub.

Re. this report: it makes sense to me now that the rendering issues are still happening on the new application, if the problem comes from Gecko and this component was in use in both old and new application.

So, I guess the best thing to do now is to re-post the old report's contents into a comment on the new one.

**Comment by j1elo** [comment]
Reposted here: https://bugzilla.mozilla.org/show_bug.cgi?id=1685629#c1
